### PR TITLE
chore(site): remove search users and groups xservice

### LINKED ISSUE #9941: Additional Batch #2 - Replace XState by react-query
Replace:

- [x] template
  - [x] searchUsersAndGroups
  - [x] templateACLService
- [x] templateVersion

Related to https://github.com/coder/coder/issues/9598

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Close https://github.com/coder/coder/issues/9941

### CODE REVIEW

File: site/src/xServices/template/searchUsersAndGroupsXService.ts [hunk 1]
```diff
@@ -1,101 +0,0 @@
-import { getGroups, getTemplateACLAvailable, getUsers } from "api/api";
-import { Group, User } from "api/typesGenerated";
-import { queryToFilter } from "utils/filters";
-import { everyOneGroup } from "utils/groups";
```
Comment (code-asher) [inline site/src/xServices/template/searchUsersAndGroupsXService.ts:4]: It looks like these utils are unused now so we could delete them.

**Review by code-asher — APPROVED** [review]
Looks and works great!

File: site/src/xServices/template/searchUsersAndGroupsXService.ts (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/xServices/template/searchUsersAndGroupsXService.ts:4]: They are tho, the queryToFilter is being used in other places and I should use the everyOneGroup... good catch!

Comment (code-asher) [reply site/src/xServices/template/searchUsersAndGroupsXService.ts:4]: Oh weird, I tested and the everyone group was showing up for me even without it.

Comment (BrunoQuaresma) [reply site/src/xServices/template/searchUsersAndGroupsXService.ts:4]: You are totally right!

**Pull request merged**